clients/admin-ui/src/features/common/EditDrawer.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Box,
+  Button,
+  Drawer,
+  DrawerBody,
+  DrawerContent,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerOverlay,
+  IconButton,
+  Text,
+  useDisclosure,
+} from "@fidesui/react";
+import ConfirmationModal from "common/ConfirmationModal";
+import { ReactNode } from "react";
+
+import { CloseSolidIcon, TrashCanSolidIcon } from "~/features/common/Icon";
+
+interface DeleteProps {
+  onDelete: () => void;
+  title: string;
+  message: ReactNode;
+}
+interface Props {
+  header: string;
+  description?: string;
+  isOpen: boolean;
+  onClose: () => void;
+  deleteProps?: DeleteProps;
+  children: ReactNode;
+  /**
+   * Associates the submit button with a form, which is useful for when the button
+   * does not live directly inside the form hierarchy
+   */
+  formId?: string;
+  /**
+   * Whether or not to include the Cancel and Save buttons
+   */
+  withFooter?: boolean;
```

Comment:
Another case of [this ](https://github.com/ethyca/fides/pull/1803#discussion_r1025745890), especially because now there are "dependent" props, where `formId` is only relevant if `withFooter` is true. An intermediate solution is using ["slots"](https://reactjs.org/docs/composition-vs-inheritance.html#:~:text=React%20elements%20like%20%3CContacts%20/%3E%20and%20%3CChat%20/%3E%20are%20just%20objects%2C%20so%20you%20can%20pass%20them%20as%20props%20like%20any%20other%20data.%20This%20approach%20may%20remind%20you%20of%20%E2%80%9Cslots%E2%80%9D%20in%20other%20libraries%20but%20there%20are%20no%20limitations%20on%20what%20you%20can%20pass%20as%20props%20in%20React.)

```tsx
<EditDrawer
  footer={
    <EditDrawerFooter onClose={} formId={} />
  }
/>
```

Reply:
👍 refactored to use slots